src/dash-table/components/ControlledTable/fragments/TableTooltip.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { PureComponent } from 'react';
+
+import Tooltip, { ITooltipProps, Arrow } from 'dash-table/components/Tooltip';
+import tooltipHelper from 'dash-table/components/tooltipHelper';
+import ReactDOM from 'react-dom';
+import { isEqual } from 'core/comparer';
+
+interface IState {
+    arrow?: Arrow;
+    cell?: any;
+}
+
+export default class TableTooltip extends PureComponent<ITooltipProps, IState> {
```

Comment:
This component is *not* standalone and is meant to be used in the context of the controlled table (hence the /ControlledTable/fragments/... location) -- in order for the table not to have to keep track of the scroll's exact position for positioning the tooltip & re-rendering uselessly, exposing a `updateBounds` method below that the CTable will call when needed (scroll, mouse over, etc.) -- this is a little against React's philosophy but keeps all the info / state in the child, making it possible to re-render the tooltip only on non state affecting user operations at the table's level.